### PR TITLE
Feat: allow editing sub project parentId so it uses our new move proj…

### LINKED ISSUE #645: Feat: Allow editing sub project parentId so it uses our new move project api
### Issue & Steps to Reproduce / Feature Request
We implemented a new [API](https://docs.env0.com/reference/projects-move-project) to support moving sub projects so now it is possible to edit the sub project parentId so the project would get moved to a different project tree...

### Expected DoD
We should allow editing the parentProjectId and call the API with the project parameters so it gets moved, deleting the parentId should send 'null' (which means the project would be changed to a root project which doesn't have a parentId)

### PR DESCRIPTION
**Pull request by TomerHeber**
…ect api

### Issue & Steps to Reproduce / Feature Request
resolves #645 

### Solution

1. Added the move API call.
2. Updated the project resource schema and the update function.
3. Added unit tests for the move API call.
4. Updated acceptance tests for project resources.
5. Updated the integration tests to cover the 'move' use case.

### CODE REVIEW

File: env0/resource_project.go [hunk 1]
```diff
@@ -110,7 +109,18 @@ func resourceProjectUpdate(ctx context.Context, d *schema.ResourceData, meta int
 	apiClient := meta.(client.ApiClientInterface)
 
 	id := d.Id()
-	var payload client.ProjectCreatePayload
+	var payload client.ProjectUpdatePayload
+
+	if d.HasChange("parent_project_id") {
+		parentProjectId := d.Get("parent_project_id").(string)
+		if parentProjectId == "" {
```
Comment (TomerHeber) [inline env0/resource_project.go:116]: This was not supported in the UI.
If it's valid to make a subproject a project (with no parent), let me know, and I will modify this.

Comment (ItamarMalka) [reply env0/resource_project.go:116]: lmk if that makes sense to you or if there are any further questions 👍

File: client/project_test.go [hunk 2]
```diff
@@ -148,4 +148,21 @@ var _ = Describe("Project", func() {
...
+		}{
+			targetProjectId,
+		}
+
+		BeforeEach(func() {
+			httpCall = mockHttpClient.EXPECT().Post("/projects/"+mockProject.Id+"/move", payload, nil).Times(1)
+			apiClient.ProjectMove(mockProject.Id, targetProjectId)
+		})
+
+		It("Should send POST request with project id and target project id", func() {})
```
Comment (ItamarMalka) [inline client/project_test.go:166]: empty case here? 🤔

File: env0/resource_project.go (hunk 1 above)
Comment (TomerHeber) [reply env0/resource_project.go:116]: makes sense. I'll fix that.

Comment (TomerHeber) [reply env0/resource_project.go]: @ItamarMalka - I fixed it: updated code, tests, and integration tests.
Now support moving a subproject to become a project.

**Review by ItamarMalka — APPROVED** [review]
QAed it manually and it worked great
LGTM, Great work 👏

**Pull request merged**